README.dev.md

```diff
@@ -79,10 +79,68 @@ During the development cycle, three main supporting branches are used:
 
 ## Making a release
 
-1. Branch from `dev` and prepare the branch for the release (e.g., removing the unnecessary dev files, fix minor bugs if necessary).
-2. [Bump the version](https://github.com/DeepRank/deeprank2/blob/dev/README.dev.md#versioning).
-3. Merge the release branch into `main` (and `dev`), and [run the tests](https://github.com/DeepRank/deeprank2/blob/dev/README.dev.md#running-the-tests).
-4. Go to https://github.com/DeepRank/deeprank2/releases and draft a new release; create a new tag for the release, generate release notes automatically and adjust them, and finally publish the release as latest. This will trigger [a GitHub action](https://github.com/DeepRank/deeprank2/actions/workflows/release.yml) that will take care of publishing the package on PyPi.
+### Automated release workflow:
+
+0. **IMP0RTANT:** Create a PR for the release branch (usually `dev`) and make sure that all checks pass!
+   - if everything goes well, this PR will automatically be closed after the draft release is created.
+1. Navigate to [Draft Github Release](https://github.com/DeepRank/deeprank2/actions/workflows/release_github.yml)
+   on the [Actions](https://github.com/DeepRank/deeprank2/actions) tab.
+2. On the right hand side, you can select the level increase ("patch", "minor", or "major") and which branch to release from.
+   - [Follow semantic versioning conventions](https://semver.org/) to chose the level increase:
+     - `patch`: when backward compatible bug fixes were made
+     - `minor`: when functionality was added in a backward compatible manner
+     - `major`: when API-incompatible changes have been made
+   - Note that you cannot release from `main` (the default shown) using the automated workflow. To release from `main`
+     directly, you must [create the release manually](#manually-create-a-release).
+3. Visit [Actions](https://github.com/DeepRank/deeprank2/actions) tab to check whether everything went as expected.
+   - NOTE: there are two separate jobs in the workflow: "draft_release" and "tidy_workspace". The first creates the draft release on github, while the second merges changes into `dev` and closes the PR.
```

Comment:
- I don't find any `draft_release` job, I think this is the one called `github_release`
- From the github workflow I see that the branch is also merged into main (during the `github_release` job) but this is done before bumping the version. Why? 
- In general, here we need a bit more details about the steps of release_github.yml for the `github_release` and the `tidy_workspace` jobs (like when there is merging to main, when the release branch is deleted and when it's not). I think that first changes are merged into main, and then they are merged back to dev from main.

Reply:
> * I don't find any `draft_release` job, I think this is the one called `github_release`

can you do check again. It used to be called "github_release", but I changed it to "draft_release". If checking locally, can you do `pull --rebase` as I had rebased this at some point (but before tagging you for review, so it should have been up to date).

>     * From the github workflow I see that the branch is also merged into main (during the `github_release` job) but this is done before bumping the version. Why?

I think the order doesn't really matter. I did it in this order in case there are conflicts, then the bump version doesn't have to happen at all. But ultimately, all these changes are done on the "local" main branch of the ci, and are only to the remote main in the step afterwards.

>     * In general, here we need a bit more details about the steps of release_github.yml for the `github_release` and the `tidy_workspace` jobs (like when there is merging to main, when the release branch is deleted and when it's not). I think that first changes are merged into main, and then they are merged back to dev from main.

Correct, I will add details about this.